android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java

```diff
@@ -179,14 +180,6 @@ public void onClick(View v) {
         return view;
     }
 
-    @Override
-    public void onDestroyView() {
-        super.onDestroyView();
-        if (mPortfolioModel != null) {
-            mPortfolioModel.clear();
-        }
-    }
-
```

Comment:
Are you able to replicate the issue in this PR? 
The bug was caused because we were emitting too many values when the NFT fragment was created (after debugging the initial loading, it emerged we were emitting the list three times). 

Now we are notifying the fragment when the list is ready (one and only one list is emitted when ready), and the UI is updated accordingly. It's much faster and easier to follow/debug/maintain.

I've also invested a fair amount of time to manual test this, but if there are extra steps to double-check I'll be happy to give it a shot. I was not able to spot any issue. And also the loading spinner now works like a charm 😁

Reply:
I think that could be something else, though Here's a demo:

Additionally, there's one more issue with the selector, I selected `Solana` and it showed solana nfts then instantly showed all networks NFTs again, seems like some cache issue.

[NFT-state-issues.webm](https://user-images.githubusercontent.com/11755381/231760078-e23d4502-5c28-4b7b-93ea-93228bf06341.webm)

Reply:
Ouch, what a mess. I'm going to investigate this. Thanks for taking the time to record a demo. Very helpful indeed!
I'll get back to you soon.

Reply:
Added back the method to clear NFT models 👉 [`10cafba` (#17967)](https://github.com/brave/brave-core/pull/17967/commits/10cafbaf10198406ddb374a8f1aade6a4970fa92)

Now it seems to work well. The entire logic still manages to behave well even when quickly changing networks many times in a row (there are some UI flashes when many requests pile up, but nothing too serious).

Let me know your thoughts, thanks.

Reply:
Cool though I think the issue is with the network selector can be reproduced because of the previous selection response.  Discuss with @jamesmudgett or @SergeyZhukovsky to confirm how critical is this and proceed accordingly.

[network-switiching-issue.webm](https://user-images.githubusercontent.com/11755381/231820893-835bd243-b7cf-4639-a15c-da98c939358d.webm)

Reply:
This is bad _:-(_ 
I was not able to end up in an inconsistent state like the one in the video (network selector points to Solana, but all the NFTs are actually loaded).

I have an idea on how to fix this, but it's something that will probably require a separate PR. I'm going to ping James, and Serg on Slack, so we can decide how to proceed.

Reply:
Filed a new issue that will be addressed soon 👉 https://github.com/brave/brave-browser/issues/29706